src/lib/FreeC/Backend/Coq/Converter/TypeDecl.hs

```diff
@@ -374,3 +385,644 @@ convertDataDecl (IR.DataDecl _ (IR.DeclIdent _ name) typeVarDecls conDecls) = do
 -- Type synonyms are not allowed in this function.
 convertDataDecl (IR.TypeSynDecl _ _ _ _)
   = error "convertDataDecl: Type synonym not allowed."
+
+-------------------------------------------------------------------------------
+-- Instance Generation                                                       --
+-------------------------------------------------------------------------------
+-- | Builds instances for all supported typeclasses.
+--   Currently, only a @Normalform@ instance is generated.
+--
+--   Suppose we have a type
+--   > data T a1 ... an = C1 a11 ... a1m1 | ... | Ck ak1 ... akmk.
```

Comment:
Surround block-level markup with blank lines.

```suggestion
--
--   > data T a1 ... an = C1 a11 ... a1m1 | ... | Ck ak1 ... akmk.
--
```

Consider to use Unicode characters for indices, ellipses, meta-variables etc. like in the documentation of other modules.

```suggestion
--
--   > data T α₁ ... αₙ = C₁ a₍₁,₁₎ ... a₍₁,ₘ₁₎ | … | Cₖ a₍ₖ,₁₎ … a₍ₖ,ₘₖ₎.
--
```